### PR TITLE
feat: Migrate lotteryUtils to typescript

### LINKED ISSUE #439: Migrate lotteryUtils to typescript
Migrate https://github.com/pancakeswap/pancake-frontend/blob/develop/src/utils/lotteryUtils.js from JS to TS

### PR DESCRIPTION
**Pull request by tzoeov**
migrated lotteryUtils to typescript
Fix #439

### CODE REVIEW

**Comment by Chef-Chungus** [comment]
@tzoeov - Can you please resolve the conflicts.

**Comment by tzoeov** [comment]
@Chef-Chungus done

File: src/utils/lotteryUtils.ts [hunk 1]
```diff
@@ -125,7 +126,7 @@ export const getTotalClaim = async (lotteryContract, ticketsContract, account) =
     const issueIndex = await lotteryContract.methods.issueIndex().call()
     const length = await getTicketsAmount(ticketsContract, account)
     // eslint-disable-next-line prefer-spread
-    const calls1 = Array.apply(null, { length }).map((a, i) => [
+    const calls1 = Array.apply(null, Array(length)).map((a, i) => [
```
Comment (hachiojidev) [inline src/utils/lotteryUtils.ts:129]: Why is this needed?

**Comment by RabbitDoge** [comment]
Duplicated with #455
